crates/ty_python_semantic/src/types.rs

```diff
@@ -946,6 +946,19 @@ impl<'db> Type<'db> {
         }
     }
 
+    /// Returns the fallback instance type that a literal is an instance of, or `None` if the type
+    /// is not a literal.
+    pub fn literal_fallback_instance(self, db: &'db dyn Db) -> Option<Type<'db>> {
+        match self {
+            Type::StringLiteral(_) | Type::LiteralString => Some(KnownClass::Str.to_instance(db)),
+            Type::BooleanLiteral(_) => Some(KnownClass::Bool.to_instance(db)),
+            Type::IntLiteral(_) => Some(KnownClass::Int.to_instance(db)),
+            Type::BytesLiteral(_) => Some(KnownClass::Bytes.to_instance(db)),
+            Type::ModuleLiteral(_) => Some(KnownClass::ModuleType.to_instance(db)),
+            _ => None,
+        }
+    }
```

Comment:
there are other promotions we could do too:
- Should class-literal types be promoted to `type[]` types?
- `Type::FunctionLiteral`, `Type::BoundMethod`, `Type::MethodWrapper`, `Type::WrapperDescriptor`, `Type::DataclassDecorator`, `Type::DataclassTransformer`: should any/all of these types be promoted to `Callable` types?
- Should heterogeneous tuple types be promoted to homogeneous tuple types? (Easy with `TupleType::homogeneous_supertype()`)

I'm not saying we necessarily should or shouldn't do any of these... but what's the framework for thinking about which ones we should do and which we shouldn't?

Reply:
> but what's the framework for thinking about which ones we should do and which we shouldn't?

Those are good examples of other things we'll need to support with full bidirectional type inference, but I don't think I'd use the mechanism from this PR for any of them. This was meant to be a much more focused stop-gap.  So tbh, the decision tree was just that this was the minimal change that had a nice diff on the ecosystem report!

Reply:
I see, that makes sense -- it could be worth recording that in a comment here?

Reply:
Done